### PR TITLE
CHANGE(BMP-611): modify logging for dotnet services

### PR DESCRIPTION
**Pull request by FilippoRezzonico**
Before this change .NET templates and examples did not perform any logging operation. This library has been modified in order to show logs that are similar to other Mia-Platform libraries. 

Picture showing the content of different types of logs using a slightly modified hello world .NET example:
log

#### Checklist

[BMP-611](https://makeitapp.atlassian.net/browse/BMP-611)

- [X] tests are included
- [ ] the documentation is updated or integrated accordingly with your changes
- [x] the changelog has been updated accordingly with your changes
- [X] you are not committing extraneous files or sensible data
- [X] added the link to the Jira task
- [X] commit message and branch name conventions

#### Notify to

@fredmaggiowski

### CODE REVIEW

File: MiaServiceDotNetLibrary/MiaServiceDotNetLibrary.csproj [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
     <PropertyGroup>
         <TargetFramework>netcoreapp3.1</TargetFramework>
-        <PackageId>MiaServiceDotNetLibrary</PackageId>
+        <PackageId>PippoServiceDotNetLibrary</PackageId>
```
Comment (fredmaggiowski) [inline MiaServiceDotNetLibrary/MiaServiceDotNetLibrary.csproj:5]:
```suggestion
        <PackageId>MiaServiceDotNetLibrary</PackageId>
```

File: MiaServiceDotNetLibrary/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,4 @@
+using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("MiaServiceDotNetLibrary.Tests")]
+[assembly: log4net.Config.XmlConfigurator(ConfigFile = "log4net.config")]
```
Comment (fredmaggiowski) [inline MiaServiceDotNetLibrary/Properties/AssemblyInfo.cs:4]:
```suggestion
[assembly: log4net.Config.XmlConfigurator(ConfigFile = "log4net.config")]

```

File: MiaServiceDotNetLibrary/MiaServiceDotNetLibrary.csproj (hunk 1 above)
Comment (FilippoRezzonico) [reply MiaServiceDotNetLibrary/MiaServiceDotNetLibrary.csproj]: My bad! Now is set to the correct PackageId!

File: MiaServiceDotNetLibrary/Properties/AssemblyInfo.cs (hunk 2 above)
Comment (FilippoRezzonico) [reply MiaServiceDotNetLibrary/Properties/AssemblyInfo.cs]: Missing newline added!

File: MiaServiceDotNetLibrary/StartupUtils.cs [hunk 3]
```diff
@@ -19,6 +21,13 @@ public class StartupUtils
 
         public static void ConfigureMiaLibraryServices(IServiceCollection services, IConfiguration configuration)
         {
+            
+            /* The line below is necessary to invoke log4net and to load its configuration,
```
Comment (fredmaggiowski) [inline MiaServiceDotNetLibrary/StartupUtils.cs:25]: use multiline comments in the form of

```
/*
 * The line below ....
 *
 */
```

Comment (FilippoRezzonico) [reply MiaServiceDotNetLibrary/StartupUtils.cs]: Done!

File: MiaServiceDotNetLibrary/StartupUtils.cs [hunk 4]
```diff
@@ -19,6 +21,15 @@ public class StartupUtils
 
         public static void ConfigureMiaLibraryServices(IServiceCollection services, IConfiguration configuration)
         {
+
+            /* 
+            * The line below is necessary to invoke log4net and to load its configuration,
```
Comment (fredmaggiowski) [inline MiaServiceDotNetLibrary/StartupUtils.cs:26]:
```suggestion
             * The line below is necessary to invoke log4net and to load its configuration, 
```

File: MiaServiceDotNetLibrary/StartupUtils.cs [hunk 5]
```diff
@@ -19,6 +21,15 @@ public class StartupUtils
 
         public static void ConfigureMiaLibraryServices(IServiceCollection services, IConfiguration configuration)
         {
+
+            /* 
+            * The line below is necessary to invoke log4net and to load its configuration, 
+            * without this line every logging functionality will be disabled.
```
Comment (fredmaggiowski) [inline MiaServiceDotNetLibrary/StartupUtils.cs:27]:
```suggestion
             * without this line every logging functionality will be disabled.
```

File: MiaServiceDotNetLibrary/StartupUtils.cs [hunk 6]
```diff
@@ -19,6 +21,15 @@ public class StartupUtils
...
+            * Follow this link to better understand how log4net configuration works:
```
Comment (fredmaggiowski) [inline MiaServiceDotNetLibrary/StartupUtils.cs:28]:
```suggestion
             * Follow this link to better understand how log4net configuration works: 
```

File: MiaServiceDotNetLibrary/StartupUtils.cs [hunk 7]
```diff
@@ -19,6 +21,15 @@ public class StartupUtils
 
         public static void ConfigureMiaLibraryServices(IServiceCollection services, IConfiguration configuration)
         {
+
+            /* 
+            * The line below is necessary to invoke log4net and to load its configuration, 
+            * without this line every logging functionality will be disabled.
+            * Follow this link to better understand how log4net configuration works: 
+            * https://logging.apache.org/log4net/release/manual/configuration.html
```
Comment (fredmaggiowski) [inline MiaServiceDotNetLibrary/StartupUtils.cs:29]:
```suggestion
             * https://logging.apache.org/log4net/release/manual/configuration.html
```

File: MiaServiceDotNetLibrary/StartupUtils.cs [hunk 8]
```diff
@@ -19,6 +21,15 @@ public class StartupUtils
...
+            */
```
Comment (fredmaggiowski) [inline MiaServiceDotNetLibrary/StartupUtils.cs:30]:
```suggestion
             */
```

File: docs/Logging.md [hunk 9]
```diff
@@ -2,37 +2,85 @@
 You can log a message to see in DevOps console. The library uses the [log4net logging system](https://logging.apache.org/log4net/).
 
 There are two types of logging:
- * Request Logging: logs every incoming request in the typical format.
+
+ * Request Logging: logs every request related message (incoming, completed and custom ones) with the typical Mia-Platform logging format.
```
Comment (fredmaggiowski) [inline docs/Logging.md:6]:
```suggestion
 * Request Logging: logs every request related message (incoming, completed and custom ones) with the standard Mia-Platform logging format.
```

File: docs/Logging.md [hunk 10]
```diff
@@ -2,37 +2,85 @@
...
 
-By default the library will generate a log for each request, representing both the incoming request and the request completion, logs are created with the *info* level and already provide useful information for later analysis or debugging. If you need more, you can add your logs.
+```csharp
+    [HttpGet]
+        public string Get()
+        {
+            Logger.Trace("trace level message");
+            Logger.Trace("trace level message with object", new CustomProps("foo"));
+            Logger.Trace(HttpContext.Request, "trace level request message with no object");
+            Logger.Trace(HttpContext.Request,"trace level request message with object", new CustomProps("foo"));
```
Comment (fredmaggiowski) [inline docs/Logging.md:36]:
```suggestion
            Logger.Trace(HttpContext.Request, "trace level request message with object", new CustomProps("foo"));
```

**Review by fredmaggiowski — CHANGES_REQUESTED** [review]

**Comment by fredmaggiowski** [comment]
LGTM

**Pull request merged**